### PR TITLE
Update links to point to datafusion.apache.org

### PR DESCRIPTION
**Pull request by phillipleblanc**
## Which issue does this PR close?

Part of #9691

## Rationale for this change

Updates the links in the repo to point to https://datafusion.apache.org instead of https://arrow.apache.org/datafusion

## Are these changes tested?

I clicked through each link as I made the change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,10 +46,10 @@ in-memory format. [Python Bindings](https://github.com/apache/datafusion-python)
...
-- [Installation](https://arrow.apache.org/datafusion/user-guide/cli/installation.html)
-- [Rust Getting Started](https://arrow.apache.org/datafusion/user-guide/example-usage.html)
-- [Rust DataFrame API](https://arrow.apache.org/datafusion/user-guide/dataframe.html)
+- [Project Site](https://datafusion.apache.org/)
+- [Installation](https://datafusion.apache.org/user-guide/cli/installation.html)
+- [Rust Getting Started](https://datafusion.apache.org/user-guide/example-usage.html)
+- [Rust DataFrame API](https://datafusion.apache.org/user-guide/dataframe.html)
 - [Rust API docs](https://docs.rs/datafusion/latest/datafusion)
 - [Rust Examples](https://github.com/apache/datafusion/tree/master/datafusion-examples)
 - [Python DataFrame API](https://arrow.apache.org/datafusion-python/)
```
Comment (phillipleblanc) [inline README.md:55]: What should the new URL be for https://arrow.apache.org/datafusion-python/ ? I'm thinking https://datafusion.apache.org/python. That change will need to be made in github.com/apache/datafusion-python first

File: dev/release/README.md [hunk 2]
```diff
@@ -463,7 +463,7 @@ svn delete -m "delete old DataFusion release" https://dist.apache.org/repos/dist
 - Checkout the `asf-site` branch
 - Copy content from `docs/build/html/*` to the `datafusion` directory in arrow-site
 - Create a PR against the `asf-site` branch ([example](https://github.com/apache/arrow-site/pull/237))
-- Once the PR is merged, the content will be published to https://arrow.apache.org/datafusion/ by GitHub Pages (this
+- Once the PR is merged, the content will be published to https://datafusion.apache.org/ by GitHub Pages (this
```
Comment (phillipleblanc) [inline dev/release/README.md:466]: This section seems outdated? I can remove it if so

File: README.md (hunk 1 above)
Comment (phillipleblanc) [reply README.md:55]: It will also need a redirect link similar to https://github.com/apache/arrow-site/pull/502

Comment (andygrove) [reply README.md:55]: Related change to datafusion-python: https://github.com/apache/datafusion-python/pull/643

File: dev/release/README.md (hunk 2 above)
Comment (alamb) [reply dev/release/README.md:466]: Yes, I agree this entire section (`### Publish the User Guide to the Arrow Site`) is outdated -- let's delete it as a follow on PR (no need to do it as part of this PR)

**Review by alamb — APPROVED** [review]
Love it -- thank you @phillipleblanc  for keeping the migration 🚋  rolling 🚀

**Pull request merged**